### PR TITLE
fix some animations

### PR DESCRIPTION
**Pull request by nobnago**

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java [hunk 1]
```diff
@@ -1381,7 +1381,7 @@ public void onPostExecute(Payload data) {
                     } else if (resultType.equals("noChanges")) {
                         // show no changes message, use false flag so we don't show "sync error" as the Dialog title
                         dialogMessage = res.getString(R.string.sync_no_changes_message);
-                        showSyncLogDialog(joinSyncMessages(dialogMessage, syncMessage), false);
+                       showSyncLogDialog(joinSyncMessages(dialogMessage, syncMessage), false);
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java]: is there something wrong with your editor? the whitespace seems to be getting changed for no good reason in this commit

Comment (nobnago) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.java]: true, was accidentally...

Am 16.03.2015 01:44 schrieb Tim Rae:

File: AnkiDroid/src/main/java/com/ichi2/compat/CompatV16.java [hunk 2]
```diff
@@ -25,5 +31,22 @@ public String detagged(String txt) {
         return Html.escapeHtml(txt);
     }
 
+    @Override
+    public void adjustSmallWidgetDimensions(RemoteViews views, int id, int left, int top, int right, int bottom, float ts) {
+        views.setTextViewTextSize(R.id.widget_due, TypedValue.COMPLEX_UNIT_SP, ts);
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/compat/CompatV16.java]: would you mind adding some comments about what's happening here and why it's not required for previous API versions?

**Comment by nobnago** [comment]
includes three issues:
- consistent animations (right/left) in switching between activities
- fix of distorted widget
- improvement of showing collapsed and not collapsed decks screenshot_2015-03-16-11-24-01

Regarding the modification of libanki: Libanki contained some parts which where necessary for AnkiDroid implementation from the beginning on. Usually, I marked them explicitly in a comment. The addition of " (+)" to mark collapsed decks was done in libanki too. However, there might be a better way of doing this...

**Comment by timrae** [comment]
Thanks, that comment was referring specifically to f48595b. The screenshot helps a lot to understand, and I just checked properly and your changes to libanki are indeed in the parts which are marked as not being in python, so no problem there.

**Comment by hssm** [comment]
@nobnago I've improved the code for building the deck list in #745 so you shouldn't need to make any additional modifications outside of the DeckPicker code that builds the UI. The new method also lets you know if you have subdecks (or which level of subdeck you are currently at), so there is no need for the additional `hasChildren` method.

I do like the look of those arrows a lot better than what we currently have.

**Comment by timrae** [comment]
OK I've merged #745 now, so this will need to be reworked. I also like the look of the arrows in this PR better :+1:

**Comment by ospalh** [comment]
If i may chip in, one problem i see with the arrows on Anki desktop is that decks with no subdecks get no marker. Look at “ 2-Phrasen” vs. “→3-Steuern” in the image above. They look like they are of a different hierarchical level when they are not.
I have patched my personal Anki to include bullets in these cases, (and changed the other two symbols) ospalh/anki@e93a5c4,

bullets

**Comment by nobnago** [comment]
Well, actually it's not a bug, it's a feature! 
For my feeling, the hierarchy is shown clearly by the indentation of the text and the symbol is just an "extra" like on Anki Desktop.
In the end, it's a matter of taste. How do the others feel about it?

**Comment by timrae** [comment]
I think it looks good how it is... I'm generally in favor of consistency
with Anki Desktop ;)

**Comment by ospalh** [comment]
> consistency with Anki Desktop

So you want to get rid of the learn count on the deck picker? (Another point where i have patched my Anki desktop, to bring *that* in line with AnkiDroid.)

**Comment by timrae** [comment]
Yeah I wouldn't be opposed to removing the learn counts from the deck
picker... one particularly vocal forum member might take strong exception
though lol

**Comment by ospalh** [comment]
Then again, i wouldn't mind too much if the bullet would not be added. I would just do it for my own version of AnkiDroid.

**Comment by timrae** [comment]
I see you've added some new commits to get it working, but it seems we're refactoring faster than you can keep up with sorry! If it's alright with you I'll rebase this myself today and put up a new PR....(?)

**Comment by timrae** [comment]
Thanks for this @nobnago it's looking really good!
I wonder how difficult it would be to turn that new arrow (which shows the subdecks) into a button which does the collapsing/expanding?

**Comment by nobnago** [comment]
I thought about that too, maybe we could solve this by three textviews instead of one...

**Comment by timrae** [comment]
Yes that would certainly work

**Comment by hssm** [comment]
I think having a separate UI element is the right thing to do since currently those symbols rely on the device rendering them how we think they should be rendered, and it could easily turn out different depending on available fonts. There was a screenshot attached to an issue not too long ago which showed the device had replaced our arrows with an emoticon and it looked pretty bad.

Although I think any major changes to the decklist might overlap with work @abvanpelt is doing. Maybe double check with each other on that.

**Comment by hssm** [comment]
[This one](https://code.google.com/p/ankidroid/issues/detail?id=2546)

**Comment by abvanpelt** [comment]
Thank you, @hssm, for the heads up! I've been keeping my eye on the pull requests and noticed some changes from here, in particular, I'll have to migrate. That's fine by me -- I'm moving pretty slowly, and these changes are too drastic so far.

**Pull request merged**